File: number.js
const prompt = require("prompt-sync")();

const playGame = ()=>{
    console.log("welcome!")
secreteNumber = Math.floor(Math.random() * 100) + 1
let tries = 0
let isGameOve = false
// console.log(secreteNumber)

let difficulty = prompt("chose your difficulty, type 'h' for Hard or 'e' for Easy\n")

if(difficulty == 'h'){
    tries = 10
}else if(difficulty =='e'){
    tries = 15
}

while( tries > 0 && isGameOve == false){

    let userGuess = +prompt("Choose a number between 1 and 100 \n")

    if(userGuess == secreteNumber){
        console.log("You win!!Thank you for playing!")
        isGameOve = true
    }
    else if (userGuess < secreteNumber){
        tries--
        console.log(`Too low, you still can try ${tries} time(s)`)
    }
    else if(userGuess > secreteNumber){
        tries--
        console.log(`Too high, you can still try ${tries} time(s)`)
    }    
    }
if(tries == 0){
    console.log("You ran out of tries, better luck next time")
}

}

while( prompt("Would you like to play a guessing game? press 'y' or 'n' \n")=='y'){
    playGame()
}

File: number.py
import random
def play_game():
    tries = 0

    secret_number = random.randint(1,100)
    #print(f"psss, the secret number is {secret_number}")
    print("Welcome! Guess the secret number")

    difficulty = input("chose your difficulty, type 'h' for Hard or 'e' for Easy\n")
    game_over = False
    if difficulty == 'h':
        tries =10
    elif difficulty == 'e':
        tries = 15

    while tries > 0 and  not game_over :

        userGuess = int(input('Guess a number between 1 and 100\n'))

        if(userGuess == secret_number):
            
            print(f"You got it!! you win!")
            game_over = True
        elif(userGuess < secret_number):
            tries -=1
            print(f"Too low, you still can try {tries}more time(s)")
        elif(userGuess> secret_number):
            tries -=1
            print(f"Too High, you still can try {tries}more time(s)")               

    if tries == 0 :
        print("You ran out of tries. Better luck next time")
        


while input("Would you like to play a guessing game? type 'y' or 'n' \n") == 'y': 
    play_game()       